Tests/DataPipeline/DataPipelineImplementation+ScreenViewsTest.swift

```diff
@@ -14,17 +14,25 @@ class DataPipelineImplementationScreenViewsTest: IntegrationTest {
     override func setUp() {
         super.setUp()
 
-        let analytics = CustomerIO.shared.analytics
+        guard let analytics = CustomerIO.shared.analytics else {
+            fatalError("Analytics instance is nil. The SDK has been set up incorrectly.")
+        }
+
         // setting up required plugins
-        outputReader = attachPlugin(analytics: analytics, plugin: OutputReaderPlugin())
-        autoTrackingScreenViews = attachPlugin(analytics: analytics, plugin: AutoTrackingScreenViews())
+        outputReader = analytics.addPluginOnce(plugin: OutputReaderPlugin())
+        autoTrackingScreenViews = analytics.addPluginOnce(plugin: AutoTrackingScreenViews())
         // wait for analytics queue to start emitting events
-        waitUntilStarted(analytics: analytics)
+        analytics.waitUntilStarted()
 
         // Screenview events are ignored if no profile identified
         CustomerIO.shared.identify(identifier: String.random)
     }
 
+    override func tearDown() {
+        CustomerIO.shared.clearIdentify()
```

Comment:
The main reason for calling `clearIdentify` is to reset `analytics` instance and its storage. I agree base `tearDown` could do more, but then this can raise more challenges impacting other tests which currently I didn't plan for this PR and the ticket.

For now, I think it is okay to keep here. But gradually, update the base test to do more and help remove boilerplate code from subclasses so it is consistent and easier to maintain.

Reply:
> but then this can raise more challenges impacting other tests which currently I didn't plan for this PR and the ticket.

I do understand that a change to a base class `tearDown` could make other existing test functions fail. Meaning that a change like this could be scope creep as other tests will need to be changed. 

If a change like resetting the SDK state in `tearDown` does indeed make existing tests break, that is a problem. Test functions should not pass or fail, depending on the result of test functions that execute before it. The fact that test functions fail indicates we have bad tests that need addressed. 

From this point, I am OK with making this change in another PR as long as a ticket is created planning on making this change.

Reply:
Agreed, created [ticket](https://linear.app/customerio/issue/MOB-50/base-unittest-and-integrationtest-classes) to tackle this